classes/output/mobile.php

```diff
@@ -166,40 +170,97 @@ public static function mobile_view_activity($args) {
+                        }
+
+                        $pagecounter++;
+                        if($pagecounter > sizeof($questionnaire['questions'])){
+                            break; //exit condition
+                        }
+                    }
+                    
+                    $i = 0;
+                    $counter = 1;
+                    foreach($data['questions'] as $dataq){
+                        
+                        foreach ($dataq as $arr) {
+                            $data['pagequestions'][$i] = $arr;
+                            $i++;
+                            $counter++;
+                            if($counter > sizeof($questionnaire['questions'])){
+                                break; //exit condition
+                            }
+                        }
+                    }
+                }
+
+                $data['prevpage'] = 0;
+                $data['nextpage'] = 0;
             }
         } else {
             $data['emptypage'] = true;
             $data['emptypage_content'] = get_string('questionnaire:submit', 'questionnaire');
         }
+
         /**
          *let each pagequestions know it's current required step, and fill up the final required step
          *logic states that we get all the required steps and give them an counter,
          *we get the final required count and check it againts the input once it's sent to a js file
          *if its the final required count we display the button
         */
-        $currentRequiredResponse = 0;
+        $currentrequiredresponse = 0;
         foreach( $data['pagequestions'] as &$pagequestion ) {
             
             if($pagequestion['info']['required'] == 'y') {
-                $currentRequiredResponse++;
-                $pagequestion['info']['current_required_resp'] = $currentRequiredResponse;
+                $currentrequiredresponse++;
+                $pagequestion['info']['current_required_resp'] = $currentrequiredresponse;
             }
         }
         //let each pagequestions know what the final required field is 
         foreach( $data['pagequestions'] as &$pagequestion ) {
-            $pagequestion['info']['final_required_resp'] = $currentRequiredResponse;
+            $pagequestion['info']['final_required_resp'] = $currentrequiredresponse;
         }
         //getting js file ready for injection... thanks JJ
-        $questionnairejs = $CFG->dirroot . '/mod/questionnaire/javascript/mobile_questionnaire.js';
-        $handle = fopen($questionnairejs, "r");
-        $questionnairejs = fread($handle, filesize($questionnairejs));
-        fclose($handle);
+        $questionnairejs = '';
+        if($pagebreaks == false) { //checking for pagebreaks
+            $questionnairejs = $CFG->dirroot . '/mod/questionnaire/javascript/mobile_questionnaire.js';
+            $handle = fopen($questionnairejs, "r");
+            $questionnairejs = fread($handle, filesize($questionnairejs));
+            fclose($handle);
```

Comment:
Found a moodle function that does this for us. Can use this on the next run. Function is called: file_get_contents()